src/main/java/com/williamfiset/algorithms/datastructures/stack/impl/ArrayStack.java

```diff
@@ -0,0 +1,62 @@
+package com.williamfiset.algorithms.datastructures.stack.impl;
+
+import com.williamfiset.algorithms.datastructures.stack.Stack;
+import java.util.Arrays;
+import java.util.EmptyStackException;
+
+/** @author liujingkun */
+public class ArrayStack<T> implements Stack<T> {
+  private int capacity;
+  private Object[] data;
+  private int size;
+
+  public ArrayStack() {
+    // the factor of 10 is the magic of speed!!
+    capacity = 10;
+    size = 0;
+    data = new Object[capacity];
+  }
+
+  @Override
+  public int size() {
+    return size;
+  }
+
+  @Override
+  public boolean isEmpty() {
+    return 0 == size;
```

Comment:
nit: It's generally easier to read left to right, so I would say size == 0 is more natural than 0 == size

Reply:
has fixed